test/unit/range/view/view_repeat_test.cpp

```diff
@@ -0,0 +1,116 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2019, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2019, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE
+// -----------------------------------------------------------------------------------------------------
+
+#include <gtest/gtest.h>
+
+#include <seqan3/core/metafunction/range.hpp>
+#include <seqan3/core/metafunction/iterator.hpp>
+#include <seqan3/io/stream/debug_stream.hpp>
+#include <seqan3/range/view/persist.hpp>
+#include <seqan3/range/view/repeat.hpp>
+#include <seqan3/range/view/take.hpp>
+#include <seqan3/range/view/take_exactly.hpp>
+#include <seqan3/std/ranges>
+
+using namespace seqan3;
+
+TEST(general, construction)
+{
+    // char
+    char chr{'A'};
+    auto v = view::repeat(chr);
+
+    EXPECT_TRUE((std::is_default_constructible_v<decltype(v)>));
+    EXPECT_TRUE((std::is_copy_constructible_v<decltype(v)>));
+    EXPECT_TRUE((std::is_move_constructible_v<decltype(v)>));
+    EXPECT_TRUE((std::is_copy_assignable_v<decltype(v)>));
+    EXPECT_TRUE((std::is_move_assignable_v<decltype(v)>));
+
+    // char const
+    char const chr_const{'A'};
+    auto v_const = view::repeat(chr_const);
+
+    EXPECT_TRUE((std::is_default_constructible_v<decltype(v_const)>));
+    EXPECT_TRUE((std::is_copy_constructible_v<decltype(v_const)>));
+    EXPECT_TRUE((std::is_move_constructible_v<decltype(v_const)>));
+    EXPECT_TRUE((std::is_copy_assignable_v<decltype(v_const)>));
+    EXPECT_TRUE((std::is_move_assignable_v<decltype(v_const)>));
+}
+
+TEST(general, concept)
+{
+    char chr{'A'};
+    auto v = view::repeat(chr);
+
+    EXPECT_TRUE((std::ranges::RandomAccessRange<decltype(v)>));
+    EXPECT_FALSE((std::ranges::ContiguousRange<decltype(v)>));
+    EXPECT_TRUE((std::ranges::View<decltype(v)>));
+    EXPECT_FALSE((std::ranges::SizedRange<decltype(v)>));
+    EXPECT_FALSE((std::ranges::CommonRange<decltype(v)>));
+    EXPECT_TRUE((std::ranges::OutputRange<decltype(v), char>));
+}
+
+TEST(general, iterator)
+{
+    auto v = view::repeat('A');
+
+    EXPECT_TRUE(v.begin() == v.begin());
+
+    EXPECT_FALSE(v.begin() == v.end());
+    EXPECT_FALSE(v.end() == v.begin());
+    EXPECT_TRUE(v.begin() != v.end());
+    EXPECT_TRUE(v.end() != v.begin());
+
+    EXPECT_FALSE(v.cbegin() == v.cend());
+    EXPECT_FALSE(v.cend() == v.cbegin());
+    EXPECT_TRUE(v.cbegin() != v.cend());
+    EXPECT_TRUE(v.cend() != v.cbegin());
+
+    auto it = v.begin();
+    EXPECT_EQ(*it, 'A');
+
+    ++it;
+    EXPECT_EQ(*it, 'A');
+    EXPECT_NE(it, v.end());
+
+    // assignment
+    *it = 'X';
+    EXPECT_EQ(*it, 'X');
```

Comment:
You should test all other operators as well. Like decrement, distance, postscript and weakly ordered operators.

Reply:
Well, I would just test the random_access_iterator since I inherit all the functionality?

Reply:
But why should someone change the RA-iterator to be **not** RA? o.O

Reply:
What's the `random_access_iterator_test` for then?

Reply:
> So you expect your class to behaves like a RAiterator, then you should test, that it in fact, it does.

I don't think this is necessary in all cases. If you have tests that verify that your Range is RandomAccessRange, you automatically verify that the iterator satisfies all syntactic requirements of RandomAccessIterator. The same holds for SizedRange and SizedSentinel behaviour of the iterator.
If you have further tests that do something like ranges::equal, you automatically check that the semantic requirements of `++` are met; and if you then also have checks for the subscript, I think you are very much on the safe side.
Yes, if someone now changes the random_access_iterator_base to change the behavior of `--` than this might not be detected here, but that's why random_access_iterator has dedicated tests that handle these situations.

Reply:
Guys, I am not talking about that someone changes the random_access_iterator_base. I am talking about someone that might (for what ever reasons) changes the iterator of the repeat_n view. And then it is clearly not tested at all; will be green in the tests, although it might be wrong because this someone else thought, everything is decently tested. 
I know that it sometimes feels over the edge, because we are trying to achieve something big companies have entire teams for. But even if we have some redundancies, this is even better. Because we stumbled more than once over things that worked in the one place but not in the other.

Reply:
fine. take a look at the test now.